src/rgw/rgw_rest_swift.cc

```diff
@@ -609,7 +615,13 @@ static int get_swift_versioning_settings(
     swift_ver_location = boost::in_place(std::string());
   }
 
-  if (s->info.env->exists("HTTP_X_VERSIONS_LOCATION")) {
+  bool ver_exists = s->info.env->exists("HTTP_X_VERSIONS_LOCATION");
+  bool his_exists = s->info.env->exists("HTTP_X_HISTORY_LOCATION");
+  if (ver_exists && his_exists) {
```

Comment:
[add this case](https://github.com/openstack/swift/blob/2.12.0/swift/common/middleware/versioned_writes.py#L41-L43)